sphinx/ext/autodoc/__init__.py

```diff
@@ -780,11 +800,15 @@ def generate(self, more_content: Any = None, real_modname: str = None,
         # reST and no starting newline is present
         self.add_line('', sourcename)
 
-        # format the object's signature, if any
-        sig = self.format_signature()
+        # format the object's signatures, if any
+        sigs = self.format_signatures()
```

Comment:
This change means `format_signature()` will be never called from `generate()` method, right? If so, subclasses of FunctionDocumenter or MethodDocumenter should implement `format_signatures()` (plural)  instead of `format_signature()` (singular). I think that is a breaking change.